## IMPORTAMOS LAS LIBRERIAS

In [11]:
import json
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt 
import seaborn as sns
import pyarrow as pa
import pyarrow.parquet as pq
import gzip

## Extracción de Datos

In [13]:
#extraccion del Archivo json
row = [] #lista vacia para ir guardando las filas
with gzip.open (r"C:\Users\ayrc2\Documentos\Proyecto Individual DPT03\Data Sets\steam_games.json.gz",'rt', encoding='utf-8') as file: #utilizo with para que el archivo se abra y cierre
    for line in file.readlines(): #bucle para ir leyendo filas y luego agregarlas a row
        data = json.loads(line)
        row.append(data)

# genero el dataframe 
games = pd.DataFrame(row)
games

publisher                                 genres   
0                   NaN                                    NaN  \
1                   NaN                                    NaN   
2                   NaN                                    NaN   
3                   NaN                                    NaN   
4                   NaN                                    NaN   
...                 ...                                    ...   
120440  Ghost_RUS Games  [Casual, Indie, Simulation, Strategy]   
120441           Sacada              [Casual, Indie, Strategy]   
120442     Laush Studio            [Indie, Racing, Simulation]   
120443         SIXNAILS                        [Casual, Indie]   
120444              NaN                                    NaN   

                        app_name                     title   
0                            NaN                       NaN  \
1                            NaN                       NaN   
2                            NaN                       NaN   
3                            NaN                       NaN   
4                            NaN                       NaN   
...                          ...                       ...   
120440            Colony On Mars            Colony On Mars   
120441  LOGistICAL: South Africa  LOGistICAL: South Africa   
120442             Russian Roads             Russian Roads   
120443       EXIT 2 - Directions       EXIT 2 - Directions   
120444               Maze Run VR                       NaN   

                                                      url release_date   
0                                                     NaN          NaN  \
1                                                     NaN          NaN   
2                                                     NaN          NaN   
3                                                     NaN          NaN   
4                                                     NaN          NaN   
...                                                   ...          ...   
120440  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
120441  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
120442  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
120443  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
120444  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                     tags   
0                                                     NaN  \
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440              [Strategy, Indie, Casual, Simulation]   
120441                          [Strategy, Indie, Casual]   
120442                        [Indie, Simulation, Racing]   
120443  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
120444  [Early Access, Adventure, Indie, Action, Simul...   

                                              reviews_url   
0                                                     NaN  \
1                                                     NaN   
2                                                     NaN   
3                                                     NaN   
4                                                     NaN   
...                                                   ...   
120440  http://steamcommunity.com/app/773640/reviews/?...   
120441  http://steamcommunity.com/app/733530/reviews/?...   
120442  http://steamcommunity.com/app/610660/reviews/?...   
120443  http://steamcommunity.com/app/658870/reviews/?...   
120444  http://steamcommunity.com/app/681550/reviews/?...   

                                                    specs price early_access   
0                                        

In [15]:
games.shape

(120445, 13)

In [14]:
games.columns

Index(['publisher', 'genres', 'app_name', 'title', 'url', 'release_date',
       'tags', 'reviews_url', 'specs', 'price', 'early_access', 'id',
       'developer'],
      dtype='object')

## Verificar y Eliminar Valores Nulos

In [16]:
nulos_por_columna=games.isnull().sum()
nulos_por_columna

publisher       96362
genres          91593
app_name        88312
title           90360
url             88310
release_date    90377
tags            88473
reviews_url     88312
specs           88980
price           89687
early_access    88310
id              88312
developer       91609
dtype: int64

In [17]:
Total_nulos= games.isnull().sum().sum()
Total_nulos

1168997

In [18]:
#Eliminamos esos valores Nulos
steam_games = games.dropna(how="all").reset_index(drop=True)
steam_games

publisher                                             genres   
0             Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]  \
1      Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
2          Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   
3                  彼岸领域                        [Action, Adventure, Casual]   
4                   NaN                                                NaN   
...                 ...                                                ...   
32130   Ghost_RUS Games              [Casual, Indie, Simulation, Strategy]   
32131            Sacada                          [Casual, Indie, Strategy]   
32132      Laush Studio                        [Indie, Racing, Simulation]   
32133          SIXNAILS                                    [Casual, Indie]   
32134               NaN                                                NaN   

                       app_name                     title   
0           Lost Summoner Kitty       Lost Summoner Kitty  \
1                     Ironbound                 Ironbound   
2       Real Pool 3D - Poolians   Real Pool 3D - Poolians   
3                       弹炸人2222                   弹炸人2222   
4                 Log Challenge                       NaN   
...                         ...                       ...   
32130            Colony On Mars            Colony On Mars   
32131  LOGistICAL: South Africa  LOGistICAL: South Africa   
32132             Russian Roads             Russian Roads   
32133       EXIT 2 - Directions       EXIT 2 - Directions   
32134               Maze Run VR                       NaN   

                                                     url release_date   
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04  \
1      http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
2      http://store.steampowered.com/app/670290/Real_...   2017-07-24   
3         http://store.steampowered.com/app/767400/2222/   2017-12-07   
4      http://store.steampowered.com/app/773570/Log_C...          NaN   
...                                                  ...          ...   
32130  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
32131  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
32134  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                    tags   
0          [Strategy, Action, Indie, Casual, Simulation]  \
1      [Free to Play, Strategy, Indie, RPG, Card Game...   
2      [Free to Play, Simulation, Sports, Casual, Ind...   
3                            [Action, Adventure, Casual]   
4                        [Action, Indie, Casual, Sports]   
...                                                  ...   
32130              [Strategy, Indie, Casual, Simulation]   
32131                          [Strategy, Indie, Casual]   
32132                        [Indie, Simulation, Racing]   
32133  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
32134  [Early Access, Adventure, Indie, Action, Simul...   

                                             reviews_url   
0      http://steamcommunity.com/app/761140/reviews/?...  \
1      http://steamcommunity.com/app/643980/reviews/?...   
2      http://steamcommunity.com/app/670290/reviews/?...   
3      http://steamcommunity.com/app/767400/reviews/?...   
4      http://steamcommunity.com/app/773570/reviews/?...   
...                                                  ...   
32130  http://steamcommunity.com/app/773640/reviews/?...   
32131  http://steamcommunity.com/app/733530/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   
32134  http://steamcommunity.com/app/681550/reviews/?...   

                         

## Verificar y Eliminar Duplicados

In [19]:
#busco y analizo los duplicados
columnas_para_verificar = ['id']
duplicados_columnas = steam_games[steam_games.duplicated(subset=columnas_para_verificar)]
duplicados_columnas

publisher               genres   
14573                                 Bethesda Softworks             [Action]  \
30961  Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   

                                             app_name   
14573                Wolfenstein II: The New Colossus  \
30961  Batman: Arkham City - Game of the Year Edition   

                                                title   
14573                Wolfenstein II: The New Colossus  \
30961  Batman: Arkham City - Game of the Year Edition   

                                                     url release_date   
14573  http://store.steampowered.com/app/612880/Wolfe...   2017-10-26  \
30961           http://store.steampowered.com/app/200260   2012-09-07   

                                                    tags   
14573  [Action, FPS, Gore, Violent, Alternate History...  \
30961  [Action, Open World, Batman, Adventure, Stealt...   

                                             reviews_url   
14573  http://steamcommunity.com/app/612880/reviews/?...  \
30961                                                NaN   

                                                   specs  price early_access   
14573  [Single-player, Steam Achievements, Full contr...  59.99        False  \
30961  [Single-player, Steam Achievements, Steam Trad...  19.99        False   

           id                                   developer  
14573  612880                               Machine Games  
30961     NaN  Rocksteady Studios,Feral Interactive (Mac)

In [20]:
steam_games[steam_games["developer"]== "Machine Games"]

publisher    genres   
12237  Bethesda Softworks  [Action]  \
13318  Bethesda Softworks  [Action]   
13894  Bethesda Softworks  [Action]   
14573  Bethesda Softworks  [Action]   
28996  Bethesda Softworks  [Action]   

                                                app_name   
12237  Wolfenstein II: The Freedom Chronicles - Seaso...  \
13318  Wolfenstein II: The Freedom Chronicles - Episo...   
13894                   Wolfenstein II: The New Colossus   
14573                   Wolfenstein II: The New Colossus   
28996                         Wolfenstein: The New Order   

                                                   title   
12237  Wolfenstein II: The Freedom Chronicles - Seaso...  \
13318  Wolfenstein II: The Freedom Chronicles - Episo...   
13894                   Wolfenstein II: The New Colossus   
14573                   Wolfenstein II: The New Colossus   
28996                         Wolfenstein: The New Order   

                                                     url release_date   
12237  http://store.steampowered.com/app/650410/Wolfe...   2017-11-07  \
13318  http://store.steampowered.com/app/624620/Wolfe...   2017-12-13   
13894          http://store.steampowered.com/app/612880/   2017-10-26   
14573  http://store.steampowered.com/app/612880/Wolfe...   2017-10-26   
28996  http://store.steampowered.com/app/201810/Wolfe...   2014-05-19   

                                                    tags   
12237                            [Gore, Action, Violent]  \
13318                                  [Action, Violent]   
13894  [Action, FPS, Gore, Violent, Alternate History...   
14573  [Action, FPS, Gore, Violent, Alternate History...   
28996  [FPS, Action, Shooter, Alternate History, Firs...   

                                             reviews_url   
12237  http://steamcommunity.com/app/650410/reviews/?...  \
13318  http://steamcommunity.com/app/624620/reviews/?...   
13894  http://steamcommunity.com/app/612880/reviews/?...   
14573  http://steamcommunity.com/app/612880/reviews/?...   
28996  http://steamcommunity.com/app/201810/reviews/?...   

                                                   specs  price early_access   
12237  [Single-player, Downloadable Content, Full con...  24.99        False  \
13318  [Single-player, Downloadable Content, Steam Ac...   9.99        False   
13894  [Single-player, Steam Achievements, Full contr...  59.99        False   
14573  [Single-player, Steam Achievements, Full contr...  59.99        False   
28996  [Single-player, Steam Achievements, Partial Co...  19.99        False   

           id      developer  
12237  650410  Machine Games  
13318  624620  Machine Games  
13894  612880  Machine Games  
14573  612880  Machine Games  
28996  201810  Machine Games

La fila duplicada 14573 es idéntica a la 13894.. por lo que podemos eliminarla

In [21]:
steam_games[steam_games["developer"]== "Rocksteady Studios,Feral Interactive (Mac)"]

publisher               genres   
1068   Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]  \
30961  Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   
31617  Warner Bros. Interactive Entertainment, Feral ...  [Action, Adventure]   

                                             app_name   
1068   Batman: Arkham City - Game of the Year Edition  \
30961  Batman: Arkham City - Game of the Year Edition   
31617  Batman: Arkham Asylum Game of the Year Edition   

                                                title   
1068   Batman: Arkham City - Game of the Year Edition  \
30961  Batman: Arkham City - Game of the Year Edition   
31617  Batman: Arkham Asylum Game of the Year Edition   

                                                     url release_date   
1068   http://store.steampowered.com/app/200260/Batma...   2012-09-07  \
30961           http://store.steampowered.com/app/200260   2012-09-07   
31617  http://store.steampowered.com/app/35140/Batman...   2010-03-26   

                                                    tags   
1068   [Action, Open World, Batman, Adventure, Stealt...  \
30961  [Action, Open World, Batman, Adventure, Stealt...   
31617  [Action, Batman, Stealth, Adventure, Third Per...   

                                             reviews_url   
1068   http://steamcommunity.com/app/200260/reviews/?...  \
30961                                                NaN   
31617  http://steamcommunity.com/app/35140/reviews/?b...   

                                                   specs  price early_access   
1068   [Single-player, Steam Achievements, Steam Trad...  19.99        False  \
30961  [Single-player, Steam Achievements, Steam Trad...  19.99        False   
31617  [Single-player, Steam Achievements, Steam Trad...  19.99        False   

           id                                   developer  
1068   200260  Rocksteady Studios,Feral Interactive (Mac)  
30961     NaN  Rocksteady Studios,Feral Interactive (Mac)  
31617   35140  Rocksteady Studios,Feral Interactive (Mac)

La fila duplicada 30961, es idéntica a la 1068; pero contiene el dato de id nulo, mientras que la fila 1068 posee el id completo 200260. Por lo que se elimina la fila duplicada 30961 

In [22]:
#eliminamos columnas con id duplicadas
duplicados_col = [14573, 30961]
steam_games= steam_games.drop(duplicados_col)

In [23]:
columnas_para_verificar = ['id']
duplicados_columnas = steam_games[steam_games.duplicated(subset=columnas_para_verificar)]
duplicados_columnas

Empty DataFrame
Columns: [publisher, genres, app_name, title, url, release_date, tags, reviews_url, specs, price, early_access, id, developer]
Index: []

In [24]:
steam_games.shape

(32133, 13)

## Transformación de datos 

In [25]:
# cIdentifico que tipos de datos tiene el dataset para saber cual debo modificar
tipos_de_datos=steam_games.dtypes
tipos_de_datos

publisher       object
genres          object
app_name        object
title           object
url             object
release_date    object
tags            object
reviews_url     object
specs           object
price           object
early_access    object
id              object
developer       object
dtype: object

EL dataset aún es un archivo json, por lo que el tipo de dato object , lo que puede ser datos anidados con formato clave - valor. Entonces a través de un bucle se analiza columna por columna y se almacenan en un diccionario vacio para luego transformarlo en un dataset y alli poder ver el tipo de datos de las columnas.

In [26]:
tipo_data = {"columna":[],"tipos_de_datos":[]} #genero un diccionario vacio para ir almacenando lo que genere el bucle

for columna in steam_games.columns: #un bucle que va recorriendo 
    tipo_data["columna"].append(columna)
    tipo_data["tipos_de_datos"].append(steam_games[columna].apply(type).unique())

analisis= pd.DataFrame(tipo_data)
analisis

columna                     tipos_de_datos
0      publisher   [<class 'str'>, <class 'float'>]
1         genres  [<class 'list'>, <class 'float'>]
2       app_name   [<class 'str'>, <class 'float'>]
3          title   [<class 'str'>, <class 'float'>]
4            url                    [<class 'str'>]
5   release_date   [<class 'str'>, <class 'float'>]
6           tags  [<class 'list'>, <class 'float'>]
7    reviews_url   [<class 'str'>, <class 'float'>]
8          specs  [<class 'list'>, <class 'float'>]
9          price   [<class 'float'>, <class 'str'>]
10  early_access                   [<class 'bool'>]
11            id   [<class 'str'>, <class 'float'>]
12     developer   [<class 'str'>, <class 'float'>]

### Transformación de la columna "release_date"

Es necesario que pueda extraerse el año del dato fecha ; por lo que se transforma la columna "release_date", generando una con el año de estreno

In [27]:
# Extraemos el año de la columna "release_date" y reemplazamos los valores nulos con "sin dato"
steam_games['Año_estreno'] = steam_games['release_date'].str.extract(r'(\d{4})').fillna('sin dato')

steam_games

publisher                                             genres   
0             Kotoshiro      [Action, Casual, Indie, Simulation, Strategy]  \
1      Making Fun, Inc.               [Free to Play, Indie, RPG, Strategy]   
2          Poolians.com  [Casual, Free to Play, Indie, Simulation, Sports]   
3                  彼岸领域                        [Action, Adventure, Casual]   
4                   NaN                                                NaN   
...                 ...                                                ...   
32130   Ghost_RUS Games              [Casual, Indie, Simulation, Strategy]   
32131            Sacada                          [Casual, Indie, Strategy]   
32132      Laush Studio                        [Indie, Racing, Simulation]   
32133          SIXNAILS                                    [Casual, Indie]   
32134               NaN                                                NaN   

                       app_name                     title   
0           Lost Summoner Kitty       Lost Summoner Kitty  \
1                     Ironbound                 Ironbound   
2       Real Pool 3D - Poolians   Real Pool 3D - Poolians   
3                       弹炸人2222                   弹炸人2222   
4                 Log Challenge                       NaN   
...                         ...                       ...   
32130            Colony On Mars            Colony On Mars   
32131  LOGistICAL: South Africa  LOGistICAL: South Africa   
32132             Russian Roads             Russian Roads   
32133       EXIT 2 - Directions       EXIT 2 - Directions   
32134               Maze Run VR                       NaN   

                                                     url release_date   
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04  \
1      http://store.steampowered.com/app/643980/Ironb...   2018-01-04   
2      http://store.steampowered.com/app/670290/Real_...   2017-07-24   
3         http://store.steampowered.com/app/767400/2222/   2017-12-07   
4      http://store.steampowered.com/app/773570/Log_C...          NaN   
...                                                  ...          ...   
32130  http://store.steampowered.com/app/773640/Colon...   2018-01-04   
32131  http://store.steampowered.com/app/733530/LOGis...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
32134  http://store.steampowered.com/app/681550/Maze_...          NaN   

                                                    tags   
0          [Strategy, Action, Indie, Casual, Simulation]  \
1      [Free to Play, Strategy, Indie, RPG, Card Game...   
2      [Free to Play, Simulation, Sports, Casual, Ind...   
3                            [Action, Adventure, Casual]   
4                        [Action, Indie, Casual, Sports]   
...                                                  ...   
32130              [Strategy, Indie, Casual, Simulation]   
32131                          [Strategy, Indie, Casual]   
32132                        [Indie, Simulation, Racing]   
32133  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
32134  [Early Access, Adventure, Indie, Action, Simul...   

                                             reviews_url   
0      http://steamcommunity.com/app/761140/reviews/?...  \
1      http://steamcommunity.com/app/643980/reviews/?...   
2      http://steamcommunity.com/app/670290/reviews/?...   
3      http://steamcommunity.com/app/767400/reviews/?...   
4      http://steamcommunity.com/app/773570/reviews/?...   
...                                                  ...   
32130  http://steamcommunity.com/app/773640/reviews/?...   
32131  http://steamcommunity.com/app/733530/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   
32134  http://steamcommunity.com/app/681550/reviews/?...   

                         

### Transformación de la columna "price"

La columna "price", posee valores nulos que deben ser reemplazados por valores numéricos de tipo flotante. Toda la columna debe ser dato de tipo flotante.

In [28]:
#Verificamos los datos que compone la columna
steam_games["price"]

0                4.99
1        Free To Play
2        Free to Play
3                0.99
4                2.99
             ...     
32130            1.99
32131            4.99
32132            1.99
32133            4.99
32134            4.99
Name: price, Length: 32133, dtype: object

In [29]:
# Creamos una función para poder reemplazar los datos string de la columna "price" por flotantes 0.0

def cambio_a_float(valor):

    if pd.isna(valor): #esta parte de la función es para el caso de que no exista ningun string o float en el campo
        return 0.0
    try:
        flotante = float(valor) #si el valor es float lo conservo
        return flotante
    except (ValueError, TypeError): #si el valor es diferente retorno 0.0
        return 0.0
    
#Aplico esta función a la columna precio
steam_games["price"] = steam_games["price"].apply(cambio_a_float)

In [30]:
#Verificamos el cambio de tipo de dato
steam_games["price"]

0        4.99
1        0.00
2        0.00
3        0.99
4        2.99
         ... 
32130    1.99
32131    4.99
32132    1.99
32133    4.99
32134    4.99
Name: price, Length: 32133, dtype: float64

### Transformación de la columna "genres"

In [31]:
steam_games["genres"][0]

['Action', 'Casual', 'Indie', 'Simulation', 'Strategy']

In [36]:
#con explode hago que la columna genres tome cada valor que tiene almacenado y lo devuelva en un campo nuevo pero conservando el mismo id
steam_games = steam_games.explode("genres")
steam_games = steam_games.dropna(subset=["genres"])
steam_games

publisher      genres             app_name                title   
0         Kotoshiro      Action  Lost Summoner Kitty  Lost Summoner Kitty  \
0         Kotoshiro      Casual  Lost Summoner Kitty  Lost Summoner Kitty   
0         Kotoshiro       Indie  Lost Summoner Kitty  Lost Summoner Kitty   
0         Kotoshiro  Simulation  Lost Summoner Kitty  Lost Summoner Kitty   
0         Kotoshiro    Strategy  Lost Summoner Kitty  Lost Summoner Kitty   
...             ...         ...                  ...                  ...   
32132  Laush Studio       Indie        Russian Roads        Russian Roads   
32132  Laush Studio      Racing        Russian Roads        Russian Roads   
32132  Laush Studio  Simulation        Russian Roads        Russian Roads   
32133      SIXNAILS      Casual  EXIT 2 - Directions  EXIT 2 - Directions   
32133      SIXNAILS       Indie  EXIT 2 - Directions  EXIT 2 - Directions   

                                                     url release_date   
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04  \
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
0      http://store.steampowered.com/app/761140/Lost_...   2018-01-04   
...                                                  ...          ...   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32132  http://store.steampowered.com/app/610660/Russi...   2018-01-04   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   
32133  http://store.steampowered.com/app/658870/EXIT_...   2017-09-02   

                                                    tags   
0          [Strategy, Action, Indie, Casual, Simulation]  \
0          [Strategy, Action, Indie, Casual, Simulation]   
0          [Strategy, Action, Indie, Casual, Simulation]   
0          [Strategy, Action, Indie, Casual, Simulation]   
0          [Strategy, Action, Indie, Casual, Simulation]   
...                                                  ...   
32132                        [Indie, Simulation, Racing]   
32132                        [Indie, Simulation, Racing]   
32132                        [Indie, Simulation, Racing]   
32133  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
32133  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   

                                             reviews_url   
0      http://steamcommunity.com/app/761140/reviews/?...  \
0      http://steamcommunity.com/app/761140/reviews/?...   
0      http://steamcommunity.com/app/761140/reviews/?...   
0      http://steamcommunity.com/app/761140/reviews/?...   
0      http://steamcommunity.com/app/761140/reviews/?...   
...                                                  ...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   

                                                   specs  price early_access   
0                                        [Single-player]   4.99        False  \
0                                        [Single-player]   4.99        False   
0                                        [Single-player]   4.99        False   
0                                        [Single-player]   4.99        False   
0                                        [Single-player]   4.99        False   
...                                                  ...    ...          ...   
32132  [Single-player, Steam Achievements, Steam Trad...   1.99        False   
32132  [Single-player, Steam Achievements, Steam Trad...   1.99        False   
32132  [Single-player, Steam Achievements, Ste

Procedo a completar los datos que se encuentran nulos o vacios, para el caso de los datos de tipo string del dataset 

In [37]:
#buscamos espacios vacios o sin datos 
steam_games.info()

<class 'pandas.core.frame.DataFrame'>
Index: 71551 entries, 0 to 32133
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   publisher     59112 non-null  object 
 1   genres        71551 non-null  object 
 2   app_name      71549 non-null  object 
 3   title         71549 non-null  object 
 4   url           71551 non-null  object 
 5   release_date  71498 non-null  object 
 6   tags          71505 non-null  object 
 7   reviews_url   71551 non-null  object 
 8   specs         70994 non-null  object 
 9   price         71551 non-null  float64
 10  early_access  71551 non-null  object 
 11  id            71551 non-null  object 
 12  developer     71201 non-null  object 
 13  Año_estreno   71551 non-null  object 
dtypes: float64(1), object(13)
memory usage: 8.2+ MB


In [38]:
#iniciamos un archivo con las columnas que tengo que reemplazar nulos
archivo_columnas=["publisher", "app_name","title","developer"]

#genero un dataframe con las columnas y sus nulos reemplazados por "Sin Datos"
nulos= steam_games[archivo_columnas].fillna("Sin Datos")

#elimino las colummas a reemplazar y concateno ambos dataframes 
steam_games = pd.concat([steam_games.drop(archivo_columnas, axis=1), nulos], axis=1)
steam_games

genres                                                url   
0          Action  http://store.steampowered.com/app/761140/Lost_...  \
0          Casual  http://store.steampowered.com/app/761140/Lost_...   
0           Indie  http://store.steampowered.com/app/761140/Lost_...   
0      Simulation  http://store.steampowered.com/app/761140/Lost_...   
0        Strategy  http://store.steampowered.com/app/761140/Lost_...   
...           ...                                                ...   
32132       Indie  http://store.steampowered.com/app/610660/Russi...   
32132      Racing  http://store.steampowered.com/app/610660/Russi...   
32132  Simulation  http://store.steampowered.com/app/610660/Russi...   
32133      Casual  http://store.steampowered.com/app/658870/EXIT_...   
32133       Indie  http://store.steampowered.com/app/658870/EXIT_...   

      release_date                                               tags   
0       2018-01-04      [Strategy, Action, Indie, Casual, Simulation]  \
0       2018-01-04      [Strategy, Action, Indie, Casual, Simulation]   
0       2018-01-04      [Strategy, Action, Indie, Casual, Simulation]   
0       2018-01-04      [Strategy, Action, Indie, Casual, Simulation]   
0       2018-01-04      [Strategy, Action, Indie, Casual, Simulation]   
...            ...                                                ...   
32132   2018-01-04                        [Indie, Simulation, Racing]   
32132   2018-01-04                        [Indie, Simulation, Racing]   
32132   2018-01-04                        [Indie, Simulation, Racing]   
32133   2017-09-02  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   
32133   2017-09-02  [Indie, Casual, Puzzle, Singleplayer, Atmosphe...   

                                             reviews_url   
0      http://steamcommunity.com/app/761140/reviews/?...  \
0      http://steamcommunity.com/app/761140/reviews/?...   
0      http://steamcommunity.com/app/761140/reviews/?...   
0      http://steamcommunity.com/app/761140/reviews/?...   
0      http://steamcommunity.com/app/761140/reviews/?...   
...                                                  ...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32132  http://steamcommunity.com/app/610660/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   
32133  http://steamcommunity.com/app/658870/reviews/?...   

                                                   specs  price early_access   
0                                        [Single-player]   4.99        False  \
0                                        [Single-player]   4.99        False   
0                                        [Single-player]   4.99        False   
0                                        [Single-player]   4.99        False   
0                                        [Single-player]   4.99        False   
...                                                  ...    ...          ...   
32132  [Single-player, Steam Achievements, Steam Trad...   1.99        False   
32132  [Single-player, Steam Achievements, Steam Trad...   1.99        False   
32132  [Single-player, Steam Achievements, Steam Trad...   1.99        False   
32133   [Single-player, Steam Achievements, Steam Cloud]   4.99        False   
32133   [Single-player, Steam Achievements, Steam Cloud]   4.99        False   

           id Año_estreno     publisher             app_name   
0      761140        2018     Kotoshiro  Lost Summoner Kitty  \
0      761140        2018     Kotoshiro  Lost Summoner Kitty   
0      761140        2018     Kotoshiro  Lost Summoner Kitty   
0      761140        2018     Kotoshiro  Lost Summoner Kitty   
0      761140        2018     Kotoshiro  Lost Summoner Kitty   
...       ...         ...           ...                  ...   
32132  610660        2018  Laush Studio        Russian Roads   
32132  610660        2018  Laush Studio        Russian Roads   
32132  6106

In [39]:
steam_games.info()

<class 'pandas.core.frame.DataFrame'>
Index: 71551 entries, 0 to 32133
Data columns (total 14 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   genres        71551 non-null  object 
 1   url           71551 non-null  object 
 2   release_date  71498 non-null  object 
 3   tags          71505 non-null  object 
 4   reviews_url   71551 non-null  object 
 5   specs         70994 non-null  object 
 6   price         71551 non-null  float64
 7   early_access  71551 non-null  object 
 8   id            71551 non-null  object 
 9   Año_estreno   71551 non-null  object 
 10  publisher     71551 non-null  object 
 11  app_name      71551 non-null  object 
 12  title         71551 non-null  object 
 13  developer     71551 non-null  object 
dtypes: float64(1), object(13)
memory usage: 8.2+ MB


## Guardamos el dataset como csv

In [51]:
guardar = "steam_games.csv"
steam_games= steam_games.to_csv(guardar, index=False)

AttributeError: 'NoneType' object has no attribute 'to_csv'